thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/NotificationOnboardingTask.java

```diff
@@ -71,20 +85,50 @@ public void run() {
     DateTime start = (DateTime) executionContext.getExecutionResult(NOTIFICATION_START);
     DateTime end = (DateTime) executionContext.getExecutionResult(NOTIFICATION_END);
 
-
+    // Get alert config from previous output
     AlertConfigDTO alertConfig = (AlertConfigDTO) executionContext.getExecutionResult(ALERT_CONFIG);
 
     SmtpConfiguration smtpConfiguration = new SmtpConfiguration();
     smtpConfiguration.setSmtpHost(taskConfigs.getString(SMTP_HOST));
     smtpConfiguration.setSmtpPort(taskConfigs.getInt(SMTP_PORT));
-    EmailResource emailResource = new EmailResource(smtpConfiguration, alertFilterFactory,
-        taskConfigs.getString(DEFAULT_ALERT_SENDER_ADDRESS), taskConfigs.getString(DEFAULT_ALERT_RECEIVER_ADDRESS),
-        taskConfigs.getString(THIRDEYE_DASHBOARD_HOST), taskConfigs.getString(PHANTON_JS_PATH), taskConfigs.getString(ROOT_DIR));
+
+    MergedAnomalyResultManager mergeAnomalyDAO = DAORegistry.getInstance().getMergedAnomalyResultDAO();
+    List<MergedAnomalyResultDTO> anomalyCandidates = mergeAnomalyDAO
+        .findByStartTimeInRangeAndFunctionId(start.getMillis(), end.getMillis(), functionId, true);
+    anomalyCandidates = AlertFilterHelper.applyFiltrationRule(anomalyCandidates, alertFilterFactory);
+    List<AnomalyResult> filteredAnomalyResults = new ArrayList<>();
+    for (MergedAnomalyResultDTO anomaly : anomalyCandidates) {
+      filteredAnomalyResults.add(anomaly);
+    }
+
+    // Set up thirdeye config
+    ThirdEyeAnomalyConfiguration thirdeyeConfig = new ThirdEyeAnomalyConfiguration();
```

Comment:
Please use a dedicated email-config object with only the fields you need. Reusing other config classes is confusing and error-prone.

Reply:
Thanks for the comment. It is fixed by introducing EmailContentFormatterConfiguration.